### ISSUE TITLE
Fit map in view does not work when tile size is 1

### ISSUE THREAD

**Issue by NworbLegin** [mapeditor/tiled#3524]
**Describe the bug**
The "Fit Map In View" function does not work when there is a high tile count. (e.g. 600 x 400 tiles at a tile width height of 8x8)

**To Reproduce**
1. From the menu select File->New->New Map...
2. Set the settings to a tile size of 8px x 8px and a Fixed number of tiles to be Width 600 and Height 400.
3. Press OK
4. The map is initially now shown filling the view
5. Press ctrl+/ or Menu->View->Fit Map In View
6. Observe that the map is not sized to fit in the view

**Expected behavior**
I would expect that any configuration of map tile size and width/height would be able to be scaled to fit within the view when Fit Map To View is actioned.

**Specifications:**
 - OS: Windows 10 x64
 - Tiled Version: 1.9.2 - but I don't think this has ever worked correctly ;-)

**Comment by eishiya** [comment]
Looking at the code, this behaviour is intentional, for performance reasons:
https://github.com/mapeditor/tiled/blob/48c4f85616fe3bc4e0d723310aa6ae38a261bb55/src/tiled/mapview.cpp#L157-L165

**Comment by NworbLegin** [comment]
Yep - I would agree that the performance hit is acceptable. Maybe this could be on a user preference setting? "Allow large fit in view"

**Comment by NworbLegin** [comment]
I have implemented this on a local build and will generate a branch and pull-request when I have time.

**Comment by eishiya** [comment]
I'm not even sure this needs to be a user preference, it seems like clutter to me. I think users working with large maps expect some performance hitches sometimes, and all this option does is make Fit Map in View not do what it says.

**Comment by NworbLegin** [comment]
Yep - I'm tempted to agree. Is the Fir Map in View ever called outside of user control? e.g. when opening a map for the first time?

**Comment by eishiya** [comment]
Not that I can find. fitInView() and fitMapInView() are only called via the action. When creating or opening new maps, Tiled seems to use 100% zoom centred on 0,0.

**Comment by NworbLegin** [comment]
Okay - Shall I amend my commit request to remove the user settings and make the feature work without restrictions?

**Comment by eishiya** [comment]
I think it would be good to hear @bjorn's opinion, perhaps there's some reason for the performance mindfulness that we've both overlooked.
